# Assignment On Measures Of Central Tendency And Dispersion


An Institution wishes to find out their student’s ability in mathematics, reading and writing skills. The Institution wants to do an exploratory study to check the following information.

#### Questions
1. Find out how many males and females participated in the test.


2. What do you think about the students' parental level of education?


3. Who scores the most on average for math, reading and writing based on


    ● Gender
    
    ● Test preparation course
    
    
4. What do you think about the scoring variation for math, reading and writing based on


    ● Gender
   
    ● Test preparation course  
   
   
5. The management needs your help to give bonus points to the top 25% of students based on their math score, so how will you help the management to achieve this.   

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_csv(r"C:\Users\illca\Downloads\StudentsPerformance.csv")
df

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

### 1.Find out how many males and females participated in the test.

In [34]:
df["gender"].value_counts()

female    518
male      482
Name: gender, dtype: int64

#### Here we can see there are 518 Females and 482 Males participated in the test.

### 2. What do you think about the students' parental level of education?


In [4]:
Edu=df.groupby(["parental level of education"]).describe()
Edu

math score                                     \
                                 count       mean        std   min    25%   
parental level of education                                                 
associate's degree               222.0  67.977477  15.108190  26.0  57.00   
bachelor's degree                118.0  69.288136  15.164117  29.0  61.00   
high school                      196.0  61.821429  15.160466   8.0  53.75   
master's degree                   59.0  70.254237  14.843544  40.0  57.00   
some college                     226.0  67.128319  14.871790  11.0  59.00   
some high school                 179.0  63.134078  16.602894   0.0  53.00   

                                               reading score             ...  \
                              50%   75%    max         count       mean  ...   
parental level of education                                              ...   
associate's degree           67.0  80.0  100.0         222.0  71.018018  ...   
bachelor's degree            68.0  79.0  100.0         118.0  73.000000  ...   
high school                  63.0  72.0   99.0         196.0  64.602041  ...   
master's degree              73.0  81.0   95.0          59.0  75.949153  ...   
some college                 68.0  76.0  100.0         226.0  69.566372  ...   
some high school             65.0  74.0   97.0         179.0  66.759777  ...   

                                          writing score                        \
                               75%    max         count       mean        std   
parental level of education                                                     
associate's degree           81.00  100.0         222.0  70.031532  14.260787   
bachelor's degree            82.75  100.0         118.0  73.381356  14.728262   
high school                  74.25   99.0         196.0  62.403061  14.165172   
master's degree              84.50  100.0          59.0  75.677966  13.730711   
some college                 80.00  100.0         226.0  69.035398  15.089911   
some high school             79.00  100.0         179.0  64.888268  15.736197   

                                                             
                              min    25%   50%   75%    max  
parental level of education                                  
associate's degree           35.0  58.25  71.5  80.0  100.0  
bachelor's degree            38.0  62.50  74.0  83.0  100.0  
high school                  15.0  52.00  64.0  73.0  100.0  
master's degree              46.0  67.00  75.0  85.0  100.0  
some college                 19.0  60.00  70.0  79.0   99.0  
some high school             10.0  54.00  66.0  77.0  100.0  

[6 rows x 24 columns]

####  Taking students ' parental level of education' based on "Math Score"

In [5]:
Math=Edu["math score"]
Math["mean"].nlargest()

parental level of education
master's degree       70.254237
bachelor's degree     69.288136
associate's degree    67.977477
some college          67.128319
some high school      63.134078
Name: mean, dtype: float64

####  Taking students ' parental level of education' based on "Reading Score"

In [6]:
Read=Edu["reading score"]
Read["mean"].nlargest()

parental level of education
master's degree       75.949153
bachelor's degree     73.000000
associate's degree    71.018018
some college          69.566372
some high school      66.759777
Name: mean, dtype: float64

####  Taking students ' parental level of education' based on "Writing Score"

In [7]:
Write=Edu["writing score"]
Write["mean"].nlargest()

parental level of education
master's degree       75.677966
bachelor's degree     73.381356
associate's degree    70.031532
some college          69.035398
some high school      64.888268
Name: mean, dtype: float64

#### Conclusion
we can see from the above result by considering the students parental level of education based on Maths, Reading and Writing Score that the students having "master's degree" as parental level has performed more.

### 3.Who scores the most on average for math, reading and writing based on
### ● Gender


In [8]:
# TO find the average we have to find the mean()

In [9]:
Gen=df.groupby(["gender"]).describe()
Gen

math score                                                       \
            count       mean        std   min   25%   50%   75%    max   
gender                                                                   
female      518.0  63.376448  16.029928   0.0  54.0  65.0  74.0  100.0   
male        482.0  68.821577  14.556411  27.0  59.0  69.0  79.0  100.0   

       reading score             ...              writing score             \
               count       mean  ...   75%    max         count       mean   
gender                           ...                                         
female         518.0  72.590734  ...  83.0  100.0         518.0  72.467181   
male           482.0  65.545643  ...  76.0  100.0         482.0  63.446058   

                                                  
              std   min   25%   50%   75%    max  
gender                                            
female  14.844842  10.0  64.0  74.0  82.0  100.0  
male    14.227225  15.0  53.0  64.0  74.0  100.0  

[2 rows x 24 columns]

#### Taking students ' Gender' based on "Math Score"

In [10]:
Genmath=Gen["math score"]
Genmath["mean"].nlargest()


gender
male      68.821577
female    63.376448
Name: mean, dtype: float64

#### Taking students ' Gender' based on "Reading Score"

In [11]:
Genread=Gen["reading score"]
Genread["mean"].nlargest()


gender
female    72.590734
male      65.545643
Name: mean, dtype: float64

#### Taking students ' Gender' based on "Writing Score"

In [12]:
Genwrite=Gen["writing score"]
Genwrite["mean"].nlargest()

gender
female    72.467181
male      63.446058
Name: mean, dtype: float64

#### Conclusion
we can see from the above result by considering the gender on average scored on Maths, Reading Score and Writing Score that the  Female students have performed more in reading and writing and male performed more in math.

### ● Test preparation course

In [13]:
Test=df.groupby(["test preparation course"]).describe()
Test

math score                                         \
                             count      mean        std   min   25%   50%   
test preparation course                                                     
completed                    358.0  69.96648  14.521847  23.0  61.0  70.0   
none                         642.0  63.78972  15.705689   0.0  54.0  64.0   

                                     reading score             ...        \
                          75%    max         count       mean  ...   75%   
test preparation course                                        ...         
completed                79.0  100.0         358.0  74.175978  ...  84.0   
none                     74.0  100.0         642.0  66.417445  ...  76.0   

                               writing score                              \
                           max         count       mean        std   min   
test preparation course                                                    
completed                100.0         358.0  74.684358  13.236412  36.0   
none                     100.0         642.0  64.457944  15.041667  10.0   

                                                   
                          25%   50%    75%    max  
test preparation course                            
completed                67.0  76.0  83.75  100.0  
none                     54.0  65.0  74.00  100.0  

[2 rows x 24 columns]

#### For Math Score

In [14]:
df.groupby(["test preparation course"])["math score"].mean()

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64

#### For Reading Score

In [15]:
df.groupby(["test preparation course"])["reading score"].mean()

test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64

#### For Writing Score

In [16]:
df.groupby(["test preparation course"])["writing score"].mean()

test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64

#### Conclusion
we can see from the above that the students who completed test preparation course have performed well on average for math,reading and writing.

### 4. What do you think about the scoring variation for math, reading and writing based on
### ● Gender


In [17]:
# In order to find scoring variation we must find the standard deviation

#### Math Score

In [39]:
df[["gender","math score"]].groupby("gender").std()

math score
gender            
female   16.029928
male     14.556411

#### Reading Score

In [38]:
df[["gender","reading score"]].groupby("gender").std()

reading score
gender               
female      14.411018
male        14.149594

#### Writing Score

In [69]:
df[["gender","writing score"]].groupby("gender").std()

writing score
gender               
female      14.844842
male        14.227225

####  
Here from the above result by considering "Gender" based on scoring variation in Math,reading and writing score, we can see that Female students have More scoring variation than male students.

### ● Test preparation course


#### Math Score

In [40]:
df[["test preparation course","math score"]].groupby("test preparation course").std()

math score
test preparation course            
completed                 14.521847
none                      15.705689

#### Reading Score

In [41]:
df[["test preparation course","reading score"]].groupby("test preparation course").std()

reading score
test preparation course               
completed                    13.537572
none                         14.608896

#### Writing Score

In [67]:
df[["test preparation course","writing score"]].groupby("test preparation course").std()

writing score
test preparation course               
completed                    13.236412
none                         15.041667

#### 
Here from the above result by considering "Test Prepation Course" based on scoring variation in Math,reading and eriting score, we can see that The variation between those who completed and none are more or less the same.

###  5. The management needs your help to give bonus points to the top 25% of 
### students based on their math score, so how will you help the management to achieve this.

In [24]:
df.shape

(1000, 8)

In [43]:
# Here we can see that their are 1000 students data so, 25% means "1000*0.25=250", so there will be 250 students.

In [65]:
Top_25=df.nlargest(250,"math score")
Top_25

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

In [27]:
# So we are able to find the to 25% of the student based on the math score, so now the management can easily give bonus to the students.

In [70]:
df.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000

In [76]:
df["math score"].quantile(0.25)

57.0

## The End.